lib/auth/clt.go

```diff
@@ -1455,7 +1455,10 @@ func (c *Client) GetSessionEvents(namespace string, sid session.ID, afterN int,
 func (c *Client) SearchEvents(fromUTC, toUTC time.Time, namespace string, eventTypes []string, limit int, startKey string) ([]events.AuditEvent, string, error) {
 	events, lastKey, err := c.APIClient.SearchEvents(context.TODO(), fromUTC, toUTC, namespace, eventTypes, limit, startKey)
 	if err != nil {
-		return nil, "", trace.Wrap(err)
+		log.WithError(err).Debug(
+			`Attempted to call SearchEvents over gRPC but received error,
+falling back to legacy API`)
+		return c.searchEventsFallback(context.TODO(), fromUTC, toUTC, namespace, eventTypes, limit, startKey)
```

Comment:
Only perform fallback if `trace.IsNotImplemented(err)`.  Check out [`auth/httpfallback.go`](https://github.com/gravitational/teleport/blob/master/lib/auth/httpfallback.go), which uses this pattern extensively to fallback to old http APIs if the required GRPC API is not implemented.